src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1290,21 +1332,23 @@ private[immutable] trait VectorPointer[+T] {
         display3 = nullSlotAndCopy(display4, (newIndex >>> 20) & 31)
         display2 = nullSlotAndCopy(display3, (newIndex >>> 15) & 31)
         display1 = nullSlotAndCopy(display2, (newIndex >>> 10) & 31)
-        display0 = nullSlotAndCopy(display1, (newIndex >>>  5) & 31, reuseDisplay0)
+        display0 = nullSlotAndCopy0(display1, (newIndex >>>  5) & 31, reuseDisplay0)
       } else {                      // level = 6
         throw new IllegalArgumentException()
       }
     }
 
 
     // USED IN DROP
-
-    private[immutable] final def copyRange[T <: AnyRef](array: Array[T], oldLeft: Int, newLeft: Int) = {
-      val elems = java.lang.reflect.Array.newInstance(array.getClass.getComponentType, 32).asInstanceOf[Array[T]]
+    private[immutable] final def copyRange[T <: AnyRef](array: Array[T], componentType: Class[_], oldLeft: Int, newLeft: Int): Array[T] = {
+      val elems = java.lang.reflect.Array.newInstance(componentType, 32).asInstanceOf[Array[T]]
```

Comment:
do we ever support primitive arrays? is there a reason not to just always create an `Array[AnyRef]`?

Reply:
the `display1`...`display5`'s are properly `Array[Array[AnyRef]]`...`Array[Array[...[AnyRef]...]]`'s, and they use this method